### PR TITLE
Olhaholiak/add security to child service

### PR DESCRIPTION
**Pull request by OlhaHoliak**

### CODE REVIEW

File: OutOfSchool/OutOfSchool.WebApi/Services/IChildService.cs [hunk 1]
```diff
@@ -10,52 +12,80 @@ namespace OutOfSchool.WebApi.Services
...
+        /// <param name="parentId">ParentId.</param>
+        /// <param name="offsetFilter">Filter to get a part of all children that were found.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.
+        /// The result is a <see cref="SearchResult{ChildDto}"/> that contains the count of all found children and a list of children that were received.</returns>
+        /// <exception cref="ArgumentNullException">If one of the parameters was null.</exception>
+        Task<SearchResult<ChildDto>> GetByParentIdOrderedByFirstName(long parentId, OffsetFilter offsetFilter);
 
         /// <summary>
-        /// Update entity.
+        /// Get children with some UsertId.
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Services/IChildService.cs:59]: Misstype `User**t**Id`

File: OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs [hunk 2]
```diff
@@ -18,172 +18,303 @@ namespace OutOfSchool.WebApi.Services
...
         }
 
         /// <inheritdoc/>
-        public async Task<IEnumerable<ChildDto>> GetAll()
+        public async Task<SearchResult<ChildDto>> GetAllWithOffsetFilterOrderedById(OffsetFilter offsetFilter)
         {
-            logger.LogInformation("Getting all Children started.");
+            logger.LogDebug($"Getting all Children started. Amount of children to take: {offsetFilter.Size}, skip first: {offsetFilter.From}.");
+
+            if (offsetFilter is null)
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs:76]: if `offsetFilter is null` you'll get NRE in previous line with `logger.LogDebug`

File: OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs [hunk 3]
```diff
@@ -18,172 +18,303 @@ namespace OutOfSchool.WebApi.Services
...
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.localizer = localizer ?? throw new ArgumentNullException(nameof(localizer));
         }
 
         /// <inheritdoc/>
-        public async Task<ChildDto> Create(ChildDto dto)
+        public async Task<ChildDto> CreateChildForUser(ChildDto childDto, string userId)
         {
-            logger.LogInformation("Child creating was started.");
+            logger.LogDebug($"Started creation of a new child with {nameof(Child.ParentId)}:{childDto.ParentId}, {nameof(userId)}:{userId}.");
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs:44]: Might be NRE if `childDto is null`

File: OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs [hunk 4]
```diff
@@ -18,172 +18,303 @@ namespace OutOfSchool.WebApi.Services
...
+                childDto.ParentId = parent.Id;
+            }
 
-            var newChild = await repository.Create(child).ConfigureAwait(false);
+            childDto.Id = 0;
 
-            logger.LogInformation($"Child with Id = {newChild?.Id} created successfully.");
+            var newChild = await childRepository.Create(childDto.ToDomain()).ConfigureAwait(false);
+
+            logger.LogDebug($"Child with Id:{newChild?.Id} ({nameof(Child.ParentId)}:{newChild.ParentId}, {nameof(userId)}:{userId}) was created successfully.");
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs:66]: Might be NRE if `newChild is null`

File: OutOfSchool/OutOfSchool.WebApi/Models/OffsetFilter.cs [hunk 5]
```diff
@@ -1,7 +1,10 @@
...
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace OutOfSchool.WebApi.Models
 {
     public class OffsetFilter
     {
+        [Range(1, long.MaxValue)]
         public int Size { get; set; } = 12;
 
         public int From { get; set; } = 0;
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Models/OffsetFilter.cs:10]: Range[0, int.MaxValue]

File: OutOfSchool/OutOfSchool.WebApi/Models/OffsetFilter.cs [hunk 6]
```diff
@@ -1,7 +1,10 @@
-﻿namespace OutOfSchool.WebApi.Models
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace OutOfSchool.WebApi.Models
 {
     public class OffsetFilter
     {
+        [Range(1, long.MaxValue)]
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Models/OffsetFilter.cs:7]: [Range(1, int.MaxValue)]

File: OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs [hunk 7]
```diff
@@ -18,172 +18,303 @@ namespace OutOfSchool.WebApi.Services
...
+            return searchResult;
         }
 
         /// <inheritdoc/>
-        public async Task<ChildDto> Update(ChildDto dto)
+        public async Task<ChildDto> UpdateChildCheckingItsUserIdProperty(ChildDto childDto, string userId)
         {
-            logger.LogInformation($"Updating Children with Id = {dto?.Id} started.");
-            this.Check(dto);
+            logger.LogDebug($"Updating the child with Id: {childDto.Id} and {nameof(userId)}: {userId} started.");
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs:201]: NRE

File: OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs [hunk 8]
```diff
@@ -18,172 +18,303 @@ namespace OutOfSchool.WebApi.Services
...
+                    logger.LogWarning($"Prevented action! User:{userId} with {nameof(Child.ParentId)}:{child.ParentId} was trying to update his child with not his own {nameof(Child.ParentId)}:{childDto.ParentId}.");
+                    childDto.ParentId = child.ParentId;
+                }
 
-                return child.ToModel();
+                var updatedChild = await childRepository.Update(childDto.ToDomain()).ConfigureAwait(false);
+
+                logger.LogDebug($"Child with Id = {updatedChild?.Id} was updated succesfully.");
+
+                return updatedChild.ToModel();
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs:236]: NRE

**Review by VadymLevkovskyi — APPROVED** [review]

**Review by DmyMi — APPROVED** [review]

**Pull request merged**